GoodThingsComein3s/ViewControllers/SignUpLoginViewController.h

```diff
@@ -6,11 +6,20 @@
 //
 
 #import <UIKit/UIKit.h>
+#import <Parse/Parse.h>
+#import "Restaurant.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
+@protocol SignUpLoginViewControllerDelegate
+-(void) addLikedRestaurantToUser:(PFUser *)currUser restaurant:(Restaurant *)restaurant;
+
+@end
+
 @interface SignUpLoginViewController : UIViewController
 
+@property (nonatomic, weak) id<SignUpLoginViewControllerDelegate> delegate;
+@property (nonatomic) Restaurant *restaurantToAddToLikes;
```

Comment:
It actually looks like `restaurantToAddToLikes` isn't needed by `SignUpLoginViewController` at all. It is set by home vc and then passed back in the delegate method with no changes, so we can remove this property and simplify the delegate method.

Moreover, [PFUser currentUser] is always available to get the current user and so doesn't have to be passed in the delegate method. Which means the delegate method can be further simplified to something like:
`- (void)signUpLoginViewControllerDidDismiss;`